packages/website-backend/src/services/real-time/SseServer.ts

```diff
@@ -0,0 +1,72 @@
+import EventEmitter from 'events';
+import type { Server, ServerResponse } from 'http';
+
+import { AddressInfo } from 'net';
+
+import { SseClient } from './SseClient.js';
+
+export class SseServer extends EventEmitter {
+  public port?: number;
+
+  #server: Server;
+  #clients = new Set<SseClient>();
+
+  constructor(server: Server) {
+    super();
+
+    this.#server = server;
+    this.#server.on('request', (_, res) => this.#handleRequest(res));
+  }
+
+  public on(
+    eventType: 'client-connected' | 'client-disconnected',
+    callback: (client: SseClient) => void
+  ): this {
+    return super.on(eventType, callback);
+  }
+
+  public emit(
+    eventType: 'client-connected' | 'client-disconnected',
+    client: SseClient
+  ): boolean {
+    return super.emit(eventType, client);
+  }
+
+  public start(): void {
+    this.#server.listen();
+    this.port = (this.#server.address() as AddressInfo).port;
+  }
+
+  public attach(res: ServerResponse) {
+    this.#handleRequest(res);
+  }
+
+  public stop(): void {
+    this.#clients.forEach((c) => {
+      c.close();
+      this.#clients.delete(c);
+    });
+  }
+
+  #handleRequest(res: ServerResponse) {
+    res.writeHead(200, {
+      'Content-Type': 'text/event-stream',
+      'Cache-Control': 'no-cache',
+      Connection: 'keep-alive',
+      'Access-Control-Allow-Origin': '*',
```

Comment:
We should set CORS properly for the dashboard

Reply:
This should be configurable as this server will be reused eventually, but good call. I guess it would be both the acceptance domain and the 'real' domain?

Reply:
It should probably be configured during deploy, since prod should only accept requests from prod and vice versa.

Reply:
I guess an environment variable would make the most sense here.

Reply:
I am not sure where the normal env. variables are set, so it won't work until I figure that out.